File: W2/lcm.c
#include <stdio.h>

int gcd(int a, int b) {
    if ( b == 0 ) {
        return a;
    }
    return gcd(b, a%b);
}

int lcm(int x, int y) {
    return x / gcd(x, y) * y;
}

int main() {
    int first, second;
    FILE *in = fopen("lcmIn.txt", "r");
    FILE *out = fopen("lcmOut.txt", "w");
    
    fscanf(in, "%d %d", &first, &second);
    fprintf(out, "%d\n", lcm(first, second));
    
    fclose(in);
    fclose(out);
    
    return 0;
}

File: PHP/lcm.php
<?php
    function gcd($a, $b) {
        if ( $b == 0 ) {
            return $a;
        }
        return gcd($b, $a % $b);
    }

    function lcm($x, $y) {
        return $x / gcd($x, $y) * $y;
    }
    
    $in = fopen("lcmIn.txt", "r");
    $out = fopen("lcmOut.txt", "w");
    
    $numbers = fscanf($in, "%d %d");
    $first = $numbers[0];
    $second = $numbers[1];
    
    fprintf($out, "%d\n", lcm($first, $second));
    
    fclose($in);
    fclose($out);
?>
